Source/ACE/Entity/Player.cs

```diff
@@ -410,10 +537,10 @@ public void SpendXp(Skill skill, uint amount)
         }
 
         //plays particle effect like spell casting or bleed etc..
-        public void PlayParticleEffect(uint effectid)
+        public void PlayParticleEffect(Effect effectId)
         {
-            var effectevent = new GameMessageEffect(this.Guid, effectid);
-            Session.Network.EnqueueSend(effectevent);
+            var effectEvent = new GameMessageEffect(this.Guid, effectId);
+            Session.Network.EnqueueSend(effectEvent);
         }
 
         /// <summary>

```

Comment:
In SpendSkillXp below, you're not taking usage-earned xp into account:
uint currentXp = chart.Ranks[Convert.ToInt32(skill.Ranks)].TotalXp;

You don't need to fix it, but you need to document it in a TODO or a known issue.